docs/docs/Best_Practices/Sensitive_Data.md

```diff
@@ -37,11 +37,46 @@ This tells the Delphix Engine to take special precautions with this password pro
 !!! note
     Removing a previously added password property from a field and running a [Data Migration](/References/Glossary.md#data-migration) will expose the password in plaintext. If this is intentional, write a migration to ensure that the new property conforms to the new schema.
 
-# Using Environment Variables For Remote Data Passing
+## Protecting Sensitive Data with Password Vaults
+
+Plugins can also leverage the password vaults configured in the Delphix engine to avoid storing sensitive data in the engine itself. In addition, vaults can rotate secrets seamlessly behind the scenes without requiring Delphix users to update those secrets in the engine. To give users the option to choose between directly entering a secret, such as a password or private key, or retrieving it from a vault, Delphix provides [pre-defined credential types](/References/Schemas.md#delphix-specific-pre-defined-types).
+
+When using these special types, the example above becomes:
+
+```json
+{
+    "type": "object",
+    "properties": {
+        "db_connectionPort": {"type": "string"},
+        "db_credentials_supplier": {
+          "$ref": "https://delphix.com/platform/api#/definitions/passwordCredentialsSupplier"
+        }
+    }
+}
+```
+
+For details on how the user can provide the information required for a property such as `db_credentials_supplier`, see the [section on pre-defined types](/References/Schemas.md#delphix-specific-pre-defined-types).
+
+At runtime, the plugin code must convert the credentials information provided by the user into an actual set of credentials that the plugin can use. To do this, the plugin must call the library function [`retrieve_credentials`](/References/Platform_Libraries.md#retrieve_credentials). For example:
+
+```python
+from dlpx.virtualization import libs
+from dlpx.virtualization.common import PasswordCredentials
+...
+@plugin.virtual.stop()
+def my_virtual_stop(virtual_source, repository, source_config):
+  credentials = libs.retrieve_credentials(virtual_source.parameters.db_credentials_supplier)
+  assert isinstance(credentials, PasswordCredentials)
+  // Now we can use the retrieved credentials.
+  ...
```

Comment:
I know this usage is covered below, but maybe a quick example of using the credentials would be good here?
For example: 
```
connect_to_dbms(credentials.username, credentials.password)
```

Reply:
Fixed